src/components/App.js

```diff
@@ -218,6 +235,15 @@ class App extends Component<{}, State> {
     }))
   }
 
+  closeRenameModal = () => {
+    this.setState({ showRenameModal: false })
+  }
+
+  discardChanges = () => {
+    this.changeFolderNames('', '')
```

Comment:
Can you help me understand what this does?

Reply:
It sounds like there are two things happening then, and I wonder if they could be separated into two functions? i.e. one to calculate folder names from notes, and one to change the name of a folder? Then we could have a `refreshFolderNames()` call in `discardChanges` as well as in `changeFolderNames` and it might be clearer.

Reply:
the refresh in changeFolderNames is a byproduct of this.state.notes being updated, so I'm not certain that a refresh function makes sense, although correct me if I'm wrong. I also think that getFolders may satisfy what you're talking about, as it just recalculates the folders, but since it isn't stateful the update won't happen inline as I want it to. Maybe a solution is to just have folders be stateful, this actually may be the cleanest fix. I was just fixing a bug that was caused by the merge, and came up with another possible solution to look at just in case. I'll push that now, let me know what you think!